### PR TITLE
Add functions to cover one half, the not encoded part, of issue 2509

### PR DESCRIPTION
**Pull request by theseion**
On behalf of @franbuehler 

Replaces https://github.com/coreruleset/coreruleset/pull/2521 for v4.1/dev.

This PR tries to cover one half of issue https://github.com/coreruleset/coreruleset/issues/2512, the not Unicode encoded part, by adding some functions that were used in the bypass.
This PR only covers the reported functions, but maybe we should have a look at other document properties and not only document.domain (https://developer.mozilla.org/en-US/docs/Web/API/Document/domain).
And the same for atob(), btoa() and alert() (https://www.w3schools.com/jsref/obj_window.asp) -> I chose the PHP file because they have been mentioned together with eval() and the enhancement of this existing file was simple and quick. But maybe we should add a separate rule with Javascript functions instead of adding them to the existing PHP file?

Nevertheless, I'm pushing this PR now, also as a concrete basis for discussion. We can still make changes.

Next, we also need a PR to cover the Unicode part of the reported bypass. I'll have a look at this too.

### CODE REVIEW

**Comment by fzipi** [comment]
@theseion There is a warning we need to fix before merging

**Comment by theseion** [comment]
Do you mean the "Github Actions syntax check"?

**Comment by theseion** [comment]
@fzipi should be fixed now.

**Review by fzipi — APPROVED** [review]
LGTM

**Comment by fzipi** [comment]
Shall we remove the `do not merge` label here?

**Comment by theseion** [comment]
Let me do a review first.

**Comment by dune73** [comment]
Would appreciate that. The three PRs with the label are now really keeping back the BB PRs and we better get going with those.

File: tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml
Comment (theseion) [inline tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933160.yaml]: Incorrect description

**Comment by theseion** [comment]
@fzipi @franbuehler LGTM.

**Pull request merged**
